azurerm/internal/services/compute/linux_virtual_machine_scale_set_resource.go

```diff
@@ -378,6 +383,9 @@ func resourceArmLinuxVirtualMachineScaleSetCreate(d *schema.ResourceData, meta i
 
 	virtualMachineProfile := compute.VirtualMachineScaleSetVMProfile{
 		Priority: priority,
+		SecurityProfile: &compute.SecurityProfile{
+			EncryptionAtHost: utils.Bool(d.Get("encryption_at_host").(bool)),
+		},
```

Comment:
I believe this'll need to be conditionally set - rather than always set - since this presumably isn't valid when reusing a "specialized" image (e.g. an image without 'sysprep' run on Windows)? the tests should confirm that

Reply:
In the meantime I fixed the tests and apparently we don't need, from what I understood, to add a condition. Wdyt ?

Reply:
I suppose what we could do it at most only set this when encryption_at_host is true - because we could not tell whether the user is explicitly setting this to false, or omitting this attribute empty